### PR TITLE
150 resolved dynamic-render-path-vulnerability

### LINKED ISSUE #150: Brakeman: Dynamic Render Path vulnerability
Results of scan:

Confidence: Medium
Category: Dynamic Render Path
Check: Render
Message: Render path contains parameter value
Code: render(template => "pages/#{params[:page]}", {})
File: app/controllers/pages_controller.rb
Line: 8

### PR DESCRIPTION
**Pull request by Janell-Huyck**
Fixes #150 

Fix Dynamic Render Path Vulnerability in PagesController

This PR addresses a security concern raised by Brakeman concerning dynamic render paths in the PagesController. Specifically, Brakeman flagged the use of params[:page] directly in a render call as a potential risk.

Changes:
Refactored PagesController to use a whitelist approach for page rendering.
Added a type check to ensure that params[:page] is a string.
Added test for the whitelist to ensure that future developers do not add a page without adding it to the whitelist.

### CODE REVIEW

File: app/controllers/pages_controller.rb
Comment (crowesn) [inline app/controllers/pages_controller.rb:1]: Maybe extract all that logic in the render call to its own function? 
e.g.
```ruby
  def show
    if valid_page?
      render template: "pages/#{safe_page}"
    else
      render file: 'public/404.html', status: :not_found
    end
  end
...

  def safe_page
    ALLOWED_PAGES.find { |page| page == params[:page].to_s }                                                            
  end
```

Otherwise I think this looks good.

**Review by crowesn — APPROVED** [review]
See inline comment and let me know what you think. It's fine to merge as is but maybe could be a little cleaner in the render call.

File: app/controllers/pages_controller.rb
Comment (Janell-Huyck) [reply app/controllers/pages_controller.rb:1]: Sure, I can do that :). Please check the updated version.

**Comment by hortongn** [comment]
Is this PR ready for @crowesn's re-review?

**Comment by crowesn** [comment]
I was holding off on this until we decided how to handle the webpack issue encountered during deploy since I wasn’t sure if we were going to be doing any reverting.. Is it okay to merge more PRs?

From: Glen Horton ***@***.***>
Date: Friday, September 15, 2023 at 1:27 PM
To: uclibs/aaec ***@***.***>
Cc: Crowe, Sean (crowesn) ***@***.***>, Mention ***@***.***>
Subject: Re: [uclibs/aaec] 150 resolved dynamic-render-path-vulnerability (PR #181)

External Email: Use Caution



Is this PR ready for @crowesn<https://github.com/crowesn>'s re-review?

**Comment by Janell-Huyck** [comment]
After Thomas's webpacker PR is merged, I will rebase if necessary and then this should be ready for re-review and merging.

**Review by crowesn — APPROVED** [review]
Looks good, merging.

**Pull request merged**
